t/129-ssl-socket.t

```diff
@@ -2618,3 +2676,410 @@ qr/\[error\] .* ngx.socket sslhandshake: expecting 1 ~ 5 arguments \(including t
 --- no_error_log
 [alert]
 --- timeout: 5
+
+
+
+=== TEST 33: setsslcert, too many arguments
+--- config
+    resolver $TEST_NGINX_RESOLVER ipv6=off;
+    location /t {
+        content_by_lua_block {
+            local sock = ngx.socket.tcp()
+            sock:settimeout(5000)
+
+            local ok, err = sock:connect("openresty.org", 443)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            ngx.say("connected: ", ok)
+
+            local ok, err = sock.setsslcert()
+        }
+    }
+
+--- request
+GET /t
+
+--- ignore_response
+--- error_log eval
+qr/\[error\] .* ngx.socket setsslcert: expecting 1 ~ 4 arguments \(including the object\), but seen 0/
+--- timeout: 5
+
+
+
+=== TEST 34: setsslcert should return error on closed connection
+--- config
+    server_tokens off;
+    resolver $TEST_NGINX_RESOLVER ipv6=off;
+    lua_ssl_trusted_certificate ../html/ca.crt;
+
+    location /t {
+        content_by_lua_block {
+            function read_file(file)
+                local f = io.open(file, "rb")
+                local content = f:read("*all")
+                f:close()
+                return content
+            end
+
+            local sock = ngx.socket.tcp()
+            sock:settimeout(3000)
+
+            local cert = read_file("$TEST_NGINX_HTML_DIR/client.crt")
+            local key = read_file("$TEST_NGINX_HTML_DIR/client.unsecure.key")
+
+            local ok, err = sock:setsslcert(cert, key)
```

Comment:
I understand the loading part that @agentzh as actually parsing the certificates (which would be solved by re-using the existing API as I already commented).

However I also think that adding the certificate to the "global" SSL_CTX is wrong, since that would affect any other SSL objects created from it (past and present), so having an API to create a new SSL_CTX and apply configurations to it would be best.